sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala

```diff
@@ -861,4 +861,17 @@ object DateTimeUtils extends SparkDateTimeUtils {
         time, timePrecision, interval, intervalEndField)
     }
   }
+
+  /**
+   * Returns the number of nanoseconds past midnight for a given timestamp in microseconds.
+   *
+   * This method uses `toJulianDay`, which splits the timestamp into (Julian day, nanoseconds in
+   * day), and returns only the nanosecond component.
+   *
+   * @param micros The timestamp in microseconds since the epoch.
+   * @return The number of nanoseconds past midnight on that day.
+   */
+  def getNanosInADay(micros: Long): Long = {
+    toJulianDay(micros)._2
```

Comment:
Could you make your implementation form the Julian calendar, please. We should follow to the Proleptic Gregorian calendar.

How about:
```suggestion
    localTimeToNanos(microsToLocalDateTime(micros).toLocalTime)
```